### ISSUE TITLE
Provided one version for different app servers

### ISSUE THREAD

**Issue by SergeyLee** [RestComm/cluster#3]
Now we have different profiles for different app servers: JBoss 7 and JBoss EAP 6.4.
Good solution for cluster is one version for different app servers.

Idea from @jaimecasero

> the idea would be to pick any version that ensures the code compiles correctly and set scope as provided
> then, dependant projects needs to ensure the correct versions are packaged into corresponding zip files
> in the end, we need to ensure compile time API compatibiltiy
> the fact different appServers use slightly different versions is not important for the compilation time behavior
> problem would be different if we had compile time API compatiblity problems
> but i think is not the issue now, any of the given versions of infinispan for different profiles/servers would cause the "cluster" project to build/compile properly
> so, in the end is about to push the concern to the proper party, since "cluster" project is appServer agnostic, then the dependant projects that know which appServer will be packaged needs to resolve infinispan dependencies version

**Comment by jaimecasero** [comment]
To complete design decision we evaluated some toher options:

1.Create more modules in cluster project to generate different cluster artifact with different dependencies
2.Define those dependencies as optional in cluster project, and make the other project explicitly define their own version
3.Exclude those dependencies in client project, and define their own version.
4.Define cluster dependencies as provided, and let client projects define their own version.

-Option 1 seems a bit awkwards for "cluster" project, since basically the code is the same no mater what Infinispan version we use in the end/runtime.
-Option 2 is misleading,since optional dependencies were not really created for this.
-Option 3 implies the hassle of excluding dependencies and explicitly set the correct version for the corresponding final runtime/server.
-Option 4 doesnt requires exluding, and the fact we set "provided" scope kind of express its really a runtime issue, more than build-time issue. Only drawback here, is that there is not much hint for dependant projects to add Infinispan dependencies to the final package so cluster project works (this is also true for the rest of options). Basically we shouldnt be packaging any infinispan dependency, since again we expect the final runtime/container/appServer to provide those libs (hence the usage of "provided" scope)
